src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureBillingProjectWizard.ts

```diff
@@ -0,0 +1,173 @@
+import { CreateNamedProjectStep } from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/CreateNamedProjectStep'
+import { StepWizard } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepWizard'
+import { validate } from 'validate.js'
+
+
+interface AzureBillingProjectWizardProps {
+  onSuccess: (string) => void
+}
+
+export const userInfoListToProjectAccessObjects = (emails: string, role: BillingRole): Array<{email: string; role: BillingRole}> => {
+  if (emails.trim() === '') {
+    return []
+  } else {
+    return _.flatten(emails.split(',').map(email => ({ email: email.trim(), role })))
+  }
+}
+
+export const AzureBillingProjectWizard = ({ onSuccess }: AzureBillingProjectWizardProps) => {
+  const [activeStep, setActiveStep] = useState<number>(1)
+  const [subscriptionId, setSubscriptionId] = useState<string>()
+  // undefined used to indicate that the user has not yet typed in the input (don't want to show error)
+  const [billingProjectName, setBillingProjectName] = useState<string|undefined>(undefined)
+  const [userEmails, setUserEmails] = useState({ emails: '', hasError: false })
+  const [ownerEmails, setOwnerEmails] = useState({ emails: '', hasError: false })
+  // undefined used to indicate that the user has not yet made a selection
+  const [addUsersOrOwners, setAddUsersOrOwners] = useState<boolean|undefined>(undefined)
+  const [managedApp, setManagedApp] = useState<AzureManagedAppCoordinates>()
+
+  const [existingProjectNames, setExistingProjectNames] = useState<string[]>([])
+  const [projectNameErrors, setProjectNameErrors] = useState<ReactNode>()
+
+  const [isBusy, setIsBusy] = useState(false)
+
+  const createBillingProject = _.flow(
+    withBusyState(setIsBusy),
+    reportErrorAndRethrow('Error creating billing project'),
+  )(async () => {
+    if (!billingProjectName) return
+    try {
+      const users = userInfoListToProjectAccessObjects(userEmails.emails, 'User')
+      const owners = userInfoListToProjectAccessObjects(ownerEmails.emails, 'Owner')
+      const members = _.concat(users, owners)
+      await Ajax().Billing
+        .createAzureProject(
+          billingProjectName, managedApp?.tenantId, subscriptionId, managedApp?.managedResourceGroupId, members
+        )
+      onSuccess(billingProjectName)
+      // No need to event success, as that is done in the onSuccess callback
+    } catch (error: any) {
+      if (error?.status === 409) {
+        setExistingProjectNames(_.concat(billingProjectName, existingProjectNames))
+        Ajax().Metrics.captureEvent(Events.billingAzureCreationProjectCreateFail, { existingName: true })
+      } else {
+        Ajax().Metrics.captureEvent(Events.billingAzureCreationProjectCreateFail, { existingName: false })
+        throw error
+      }
+    }
+  })
+
+  useEffect(() => {
+    let isMounted = true // Necessary to avoid React warning about updates in unmounted component after project created.
```

Comment:
If we update react to v18 (🤞), the warning will go away without this

Reply:
Good to know!